### ISSUE TITLE
problem with migration from res to bsc32

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#962]
In GitLab by @vsicardi on Feb 27, 2023, 14:10

Hello @dbeltrankyl and @kinow,

I use `autosubmit  v3.14.0` to migrate   4 experiments, from res account **cns82886 to bsc32774**.

 I started with **a5cx.**

I am following the instructions at https://earth.bsc.es/gitlab/es/autosubmit/-/wikis/migrate-experiments, nevertheless the step  `(User-Target) autosubmit migrate <expid> -p --onlyremote` gave me the following error:

```
[ERROR] Command rsync --timeout=3600 --bwlimit=20000 -aq --remove-source-files /gpfs/scratch/bsc32/bsc32774/a5cx_TMP/a5cx /gpfs/scratch/bsc32/bsc32774 in mn1.bsc.es warning: rsync: link_stat "/gpfs/scratch/bsc32/bsc32774/a5cx_TMP/a5cx" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1165) [sender=3.1.0]
[eCode=6006]
[ERROR] The files/dirs on /gpfs/scratch/bsc32/bsc32774/a5cx_TMP/a5cx cannot be copied to /gpfs/scratch/bsc32/bsc32774/a5cx.[eCode=6012]
 [CRITICAL] Unable to pickup all platforms, the non-moved files are on the TEMP_DIR
 You can try again with autosubmit a5cx -p --onlyremote [eCode=7012]
```

I followed all the instructions and I created the folder /**gpfs/scratch/bsc32/bsc32774/a5cx_TMP/**,  why is it searching for the folder
 **gpfs/scratch/bsc32/bsc32774/a5cx_TMP/a5cx** ?

I would appreciate a lot if you could help me in understanding which step I can have missed.

thank you

FYI @eferre1

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 27, 2023, 14:43

Hello @vsicardi ,


> I followed all the instructions, and I created the folder /**gpfs/scratch/bsc32/bsc32774/a5cx_TMP/**, why is it searching for the folder **gpfs/scratch/bsc32/bsc32774/a5cx_TMP/a5cx** ?

I don't see anything wrong with this, what does the offer is to put the entire folder a5cx on the TEMP_DIR path, so it is searching for the correct folder.

The thing is that this folder is missing, and thus, autosubmit can't recover anything. 

I see that `/gpfs/scratch/bsc32/bsc32774/a5cx` exists tho, so maybe you did perform the command twice? you can also check if there is data left in the original user

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 27, 2023, 14:45

For example, in a4sv, this folder is still in the tmp folder. 
`/gpfs/scratch/bsc32/bsc32774/TMP_a4sv/a4sv`

**Comment by kinow** [comment]
In GitLab by @vsicardi on Feb 27, 2023, 14:48

Hi,

thank you so much for the quick reply.

`/gpfs/scratch/bsc32/bsc32774/a5cx` exists, but it has only  the logs of earthdiagnostics. It was created during the run, not now with the migrate command.

Maybe the "offer" command did not work? What should I expect after running `autosubmit migrate <expid> -o --onlyremote`

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 27, 2023, 15:01

Checking the logs, it seems the offer step didn't find data. 

```
2023-02-27 13:43:30,208 Moving remote files/dirs on transfer_node
2023-02-27 13:43:52,448 File /gpfs/scratch/cns82/cns82886/a5cx/LOG_a5cx doesn't exists 
2023-02-27 13:43:52,448 No data found in /gpfs/scratch/cns82/cns82886/a5cx for [transfer_node]
```
Could you check if there is data on there or if it was the correct user?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 27, 2023, 15:03

What you should expect is that:

* `-o` moves all data from `$scratch/$original_project/$original_user/$expid` to `$tmp_dir/$expid`
* and `-p` transfers all data `$tmp_dir/$expid` to `$scratch/$project_to/$user_to/$expid`

**Comment by kinow** [comment]
In GitLab by @vsicardi on Feb 27, 2023, 15:13

Hi,

yes, I checked and /gpfs/scratch/cns82/cns82886/a5cx/LOG_a5cx exists, as long as all the other directories in /gpfs/scratch/cns82/cns82886/a5cx.

Can the problem be related to the transfer node?

would you mind to check my platform.conf, please?

/esarchive/autosubmit/a5cx/conf/platforms_a5cx.conf

**Comment by kinow** [comment]
In GitLab by @vsicardi on Feb 27, 2023, 15:17

Then the command  offer did nothing, since `$tmp_dir/$expid` is empty.

**Comment by kinow** [comment]
In GitLab by @vsicardi on Feb 28, 2023, 11:48

Hi @dbeltrankyl,

Unfortunately, the migration is still giving problems.

I tried the last changes you suggested, but they are not working. The last thing I tried was moving TEMP_DIR to in marenostrum session instead of in the transfer node session. But it did not work.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 28, 2023, 12:01

Hello,


```
Invalid Configuration:
Invalid TEMP_DIR, Parameter must be present even if empty in [transfer_node] [eCode=7014]
2023-02-28 11:42:31,227 More info at https://autosubmit.readthedocs.io/en/v3.14.0/troubleshooting/error-codes.html 
```


It seems that TEMP_DIR is missing in transfer_node. Please fill it as empty.

`TEMP_DIR = `

**Comment by kinow** [comment]
In GitLab by @vsicardi on Feb 28, 2023, 12:45

Hi,

done, but the problem is the same:

```
Files/dirs on nord3v2 have been successfully offered
Project in platform configuration file successfully updated to cns82
Host in platform configuration file successfully updated to mn1.bsc.es
Converting the absolute symlinks into relatives on platform marenostrum4 
Moving remote files/dirs on marenostrum4
No data found in /gpfs/scratch/cns82/cns82886/a5cx for [marenostrum4]

Files/dirs on marenostrum4 have been successfully offered
Project in platform configuration file successfully updated to cns82
Host in platform configuration file successfully updated to mn-dt
Files/dirs on transfer_node have been successfully offered
The experiment has been successfully offered.
```


The TEMP_DIR is still empty, and the folder /gpfs/scratch/cns82/cns82886/a5cx is still full. 

Can I manually make a copy of the folder using the dt machine?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 28, 2023, 13:03

Weird. I don't know why it doesn't detect that there is data in that folder.

* Does the /gpfs/scratch/cns82/cns82886/a5cx/LOG_a5cx folder exists? If it doesn't it may be the source of the issue as Autosubmit has to send the convertlink.sh script there. 

* Maybe it is a permissions issue, Can you create a dummy folder in /gpfs/scratch/bsc32/bsc32774/a5cx_tmp as **CNS82886** ? (I think this is the issue as I see that it has no Write permissions for the group) 

```
bsc32070@login2:/gpfs/scratch/bsc32/bsc32774/a5cx_TMP> ll -hart
total 65K
drwxr-xr-x  2 bsc32774 bsc32 4.0K Feb 27 14:29 .
drwxr-xr-x 44 bsc32774 bsc32  32K Feb 28 12:49 ..
```
can you perform chmod -R 777 to `/gpfs/scratch/bsc32/bsc32774/a5cx_TMP` ? and then repeat the migrate offer with the corrected values 


Manually it will crash if it is not done in a specific way, I can provide you the steps to follow but I would do it as a last resort. First, please check the permissions and try again.

Edit: For autosubmit, it is missing a check permissions function and showing a different error message to address a future misunderstanding

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 1, 2023, 09:32

Hi,

Thank you very much for the support!

We are making progress.

Now everything is copied in `/gpfs/scratch/bsc32/bsc32774/a5cx_TMP`, but I get this error in the second step:

`autosubmit migrate a5cx -p`


```
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
Traceback (most recent call last):
  File "/shared/earth/software/autosubmit/3.14.0-foss-2015a-Python-2.7.9/bin/autosubmit", line 4, in <module>
    __import__('pkg_resources').run_script('autosubmit==3.14.0', 'autosubmit')
  File "/shared/earth/software/Python/2.7.9-foss-2015a/lib/python2.7/site-packages/pkg_resources/__init__.py", line 654, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/shared/earth/software/Python/2.7.9-foss-2015a/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1434, in run_script
    exec(code, namespace, namespace)
  File "/shared/earth/software/autosubmit/3.14.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.14.0-py2.7.egg/EGG-INFO/scripts/autosubmit", line 31, in <module>
    from autosubmit.autosubmit import Autosubmit
  File "/shared/earth/software/autosubmit/3.14.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.14.0-py2.7.egg/autosubmit/autosubmit.py", line 36, in <module>
    from monitor.monitor import Monitor
  File "/shared/earth/software/autosubmit/3.14.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.14.0-py2.7.egg/autosubmit/monitor/monitor.py", line 44, in <module>
    from diagram import create_bar_diagram
  File "/shared/earth/software/autosubmit/3.14.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.14.0-py2.7.egg/autosubmit/monitor/diagram.py", line 22, in <module>
    import matplotlib as mtp
  File "/shared/earth/software/matplotlib/1.5.3-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/matplotlib-1.5.3-py2.7-linux-x86_64.egg/matplotlib/__init__.py", line 1138, in <module>
    rcParams = rc_params()
  File "/shared/earth/software/matplotlib/1.5.3-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/matplotlib-1.5.3-py2.7-linux-x86_64.egg/matplotlib/__init__.py", line 971, in rc_params
    fname = matplotlib_fname()
  File "/shared/earth/software/matplotlib/1.5.3-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/matplotlib-1.5.3-py2.7-linux-x86_64.egg/matplotlib/__init__.py", line 786, in matplotlib_fname
    cwd = os.getcwdu()
OSError: [Errno 2] No such file or directory
```

Any clue?

Thank you

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 1, 2023, 09:33

FYI @acarreri

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 1, 2023, 09:35

This is related to the current path where you're launching the command. 

If you did migrate -o without --onlyremote flag, the local folder is now archived until you do the pickup. 

If you cd to your home folder or elsewhere and perform the migrate -p it should work

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 1, 2023, 10:19

Hi,

The experiment a5cx has been migrated successfully!

@dbeltrankyl Thank you very much for your support and patience!

I start the migration of my other 3 experiments, and then I will close the issue.

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 1, 2023, 16:53

Thanks @vsicardi 

Thank you for being so patient too, and it's a good use case to improve the error system and provide more hints to the user, as @dbeltrankyl is saying.

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 1, 2023, 16:53

@dbeltrankyl could you create issues for

* Edit: For autosubmit, it is missing a check permissions function and showing a different error message to address a future misunderstanding

* `OSError: [Errno 2] No such file or directory` (it would be good that it gives a hint about the possibility of the local dir being archived)

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 2, 2023, 12:00

Hi @dbeltrankyl,

the migration went smooth for 3 out of 4 experiments.

The experiment with problem is **a5fw**. Please note that the experiment's owner is rbernard(bsc32739), but I have full access to his account.

I left migrating it last night, it happened that it did the first step, the offer, but it did not complete the pick-up, and now when I relaunch it (autosubmit migrate a5fw -p), it does not find the experiment: 

```
Autosubmit is running with 3.14.0
Migrating experiment a5fw
Moving local files/dirs
[ERROR] Experiment a5fw is not archived
 [CRITICAL] The experiment cannot be picked up [eCode=7012]
More info at https://autosubmit.readthedocs.io/en/v3.14.0/troubleshooting/error-codes.html 
```

indeed this experiment is not archived in autosubmit/2022, nor in 2023 and it is not anymore on the res account. It is in the a5fw_TMP folder in bsc32739. 

Is there anything I could do to complete the migration?

Thank you very much!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 2, 2023, 12:53

Hello @vsicardi ,

Yes, it is because the experiment exists on `/esarchive/autosubmit/a5fw`

You can use

`autosubmit migrate -p --onlyremote a5fw` to finish the migration, it will ignore the local folder and just pickup the remote part

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 2, 2023, 15:41

Hi @dbeltrankyl,

thank you very much. It worked as you suggested.

Now I launched one of the migrated experiment and I have a permission error at CLEAN, as you can see in the following log:

`/esarchive/autosubmit/a5fx/tmp/LOG_a5fx/a5fx_16380101_fc0_55_CLEAN.20230302152452.out`

would you mind to check what is wrong?

The output folder has the right permissions and it belongs to the group Earth (as far as I can see)

`drwxrwsrwx 5 cns82399 Earth 4096 Dec 19 02:21 original_files`

thanks

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 2, 2023, 16:02

Hello, 

This seems related to the template itself, I guess @eferre1 can help you better

**Comment by kinow** [comment]
In GitLab by @eferre1 on Mar 2, 2023, 16:07

Hi, I just checked and it is due to the `dtsgrsync` command, that it is trying to change the permissions of the folders and since those were created with a RES account, it fails.

I checked that you are already using a branch derived from the 3.3.3.2 TAG, so it has the command integrated and this kind of errors after a transfer shouldn't happen (the command was added for this reason among others), so I'm not sure why it is still failing. I mean, it is failing because it is trying to change the permissions of a folder from another user, but it shouldn't happen since it shouldn't need to modify these permissions or the migrate should have transferred these folders too. I'm looking into it.

**Comment by kinow** [comment]
In GitLab by @eferre1 on Mar 7, 2023, 09:40

Ok, I have checked this and the files should be transferred, and if you look into the `original_files/cmorfiles` folder, the outputs from the chunk 55 are present (as an example, `tos_Omon_EC-Earth3-CC_piControl_r1i1p1f1_gn_169201-169212.nc` is there). So the problem is that the files are being transferred, but the `dtsgrsync` command is trying to change the permissions too and that it cannot do due to the migration, and then the `.err` log from the command says it, and the check that we have on the clean to be sure that the command worked is that the `.err` log is empty.

I'm not sure about how to fix this in the code, any idea @mcastril? Or if it could be done something to migrate the folders in `esarchive/exp` too in the autosubmit part, @dbeltrankyl?

A solution for this particular case is easy, to move the `original_files` folder and `rsync` it back with the bsc32 account so that it is the owner of the folders and it doesn't fail. The exact sequence of commands would be this (from the bsc32 account, better from the DT nodes):

```
cd /gpfs/archive/bsc32/esarchive/exp/ecearth/a5fx
mv original_files original_files_bckp
dtrsync -avPru original_files_bckp/ original_files/
```

Once it is done, you can check with an `diff -rq original_files_bckp original_files` that there are no differences between the folders (if there is an issue within a file, the `dtsgrsync` should pick it up, you can check its logs too), and then you can continue the run and delete the backup.

PS: I fixed the commands to run from the DT (missing `archive` path in `cd` and `/` in the `dtrsync`).

**Comment by kinow** [comment]
In GitLab by @acarreri on Mar 7, 2023, 14:42

Hi,

maybe I'm adding noise here, sorry about that, but I think we already encountered this issue: https://earth.bsc.es/gitlab/es/auto-ecearth3/-/issues/1817

**Comment by kinow** [comment]
In GitLab by @acarreri on Mar 7, 2023, 14:48

mentioned in issue auto-ecearth3#1872

**Comment by kinow** [comment]
In GitLab by @eferre1 on Mar 7, 2023, 16:31

Yes, it is true that it is the same permissions error, thanks @acarreri, I'll reopen that issue and will investigate about the `dtsgrsync` options, I don't remember why I closed it though. :sweat_smile:

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 7, 2023, 17:58

@eferre1 I think I need more details about what's the actual issue. Do you have the `.err` log from the `dtsgrsync`? And what are the details about the files in `/esarchive`?

**Comment by kinow** [comment]
In GitLab by @eferre1 on Mar 8, 2023, 08:14

> Do you have the `.err` log from the `dtsgrsync`

Yes, not exactly the `.err` log, but in the log mentioned above (in https://earth.bsc.es/gitlab/es/autosubmit/-/issues/962#note_202055), there is the content of that error log from this point onward:

`the dtsgrsync_3797198.err  is not empty`

And all the error messages are due to `rsync failed to set permissions` and nothing related to file transfers.

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 13, 2023, 09:09


Hello again,

I have a last problem with another exp **a5hn** from **cns60057 to bsc32774**. It is not fundamental to migrate it, I could start it from zero, but since it is almost migrated, I would like to ask for help. Having said that, if making it work implies too much effort, we do nothing with it. 

The offer went fine, but for the pick-up I have this message:



```
Copying remote files/dirs on marenostrum4
Copying from /gpfs/scratch/bsc32/bsc32774/a5hn_TMP/a5hn to /gpfs/scratch/bsc32/bsc32774/a5hn
Rsync launched 1 times. Can take up to 150 retrials or until all data is transfered
[ERROR] Command rsync --timeout=3600 --bwlimit=20000 -aq --remove-source-files /gpfs/scratch/bsc32/bsc32774/a5hn_TMP/a5hn /gpfs/scratch/bsc32/bsc32774 in mn1.bsc.es warning: rsync: [sender] write error: Broken pipe (32)
rsync error: error in socket IO (code 10) at io.c(846) [sender=3.1.0]
[eCode=6006]
Empty dirs on /gpfs/scratch/bsc32/bsc32774/a5hn_TMP have been successfully deleted
Files/dirs on marenostrum4 have been successfully picked up
Empty dirs on /gpfs/scratch/bsc32/bsc32774/a5hn_TMP have been successfully deleted
Files/dirs on transfer_node have been successfully picked up
The experiment has been successfully picked up.
```


And indeed the files are still in a5hn_TMP and the created links are broken

I tried also with --onlyremote 

Any suggestions?
thank you very much

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 13, 2023, 09:19

Hello, 

I think it crashed due to the experiment being 2.8T. If you keep trying with --only-remote, eventually, all data will be transferred. 

Autosubmit will resume the transfer. This should be done automatically, but the improvements made weren't enough. I'll revise it for 3.15.0b alongside the permissions issue.

> And indeed the files are still in a5hn_TMP, and the created links are broken

The links are broken because that data is still not transferred

**Comment by kinow** [comment]
In GitLab by @vsicardi on Mar 13, 2023, 09:22

Thank you @dbeltrankyl,

I keep trying, then.

Eventually, could I transfer each folder from  a5hn_TMP to a5hn, launching a manual rsync?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 13, 2023, 09:29

Well, what Autosubmit does is:

Each iteration

`rsync --timeout=3600 --bwlimit=20000 -aq --remove-source-files /gpfs/scratch/bsc32/bsc32774/a5hn_TMP/a5hn /gpfs/scratch/bsc32/bsc32774`

`find /gpfs/scratch/bsc32/bsc32774/a5hn_TMP/a5hn -depth -type d -empty -delete`

and when all data is transferred:

`chmod 755 -R /gpfs/scratch/bsc32/bsc32774/a5hn`

Also, I saw that the TEMP_DIR is filled in mn4. If you fill it in transfer_data, It will use the dt01 instead of the login nodes, and it will be less prone to issues

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 13, 2023, 09:31

And autosubmit is not iterating because the error code: `rsync error: error in socket IO (code 10) at io.c(846) [sender=3.1.0]` before we were getting "broken pipe" or "warning: rsync" not sure why the message changed but we will address it on 3.15.0b

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 13, 2023, 09:41

mentioned in commit 10095eabd6214f9b55f87a1533259d6f981d9507

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 16, 2023, 16:23

mentioned in commit 3c0473075d5eebcca26d29d07bfc016af9143f80

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 27, 2023, 09:27

I applied a fix in 4.0.0b and 3.15.0 is pending testing
